lib/packages/docutils/rst.nim

```diff
@@ -1937,13 +1948,34 @@ proc parseEnumList(p: var RstParser): PRstNode =
     wildToken: array[0..5, int] = [4, 3, 3, 4, 3, 3]  # number of tokens
     wildIndex: array[0..5, int] = [1, 0, 0, 1, 0, 0]
       # position of enumeration sequence (number/letter) in enumerator
-  result = newRstNodeA(p, rnEnumList)
   let col = currentTok(p).col
   var w = 0
   while w < wildcards.len:
     if match(p, p.idx, wildcards[w]): break
     inc w
   assert w < wildcards.len
+  proc checkAfterNewline(p: RstParser, report: bool): bool =
+    let j = tokenAfterNewline(p, start=p.idx+1)
+    if p.tok[j].kind notin {tkIndent, tkEof} and
```

Comment:
getting there, it's almost perfect...

another regression:


```
1. hello 1
1. hello 2
   foo 2
1. hello 3
   foo 3
   1. sub1
   1. sub2
   1. sub3
   zook5
1. hello 4
1. hello 5
1. hello 6
 zook6
1. hello 7

1. hello 8


1. hello 9

1. hello 10

separator

1. hello 11
1. hello 12
```

## before PR
![image](https://user-images.githubusercontent.com/2194784/110704087-5f1b1a80-81a9-11eb-970a-cc5949208b62.png)

## after PR
![image](https://user-images.githubusercontent.com/2194784/110704146-6d693680-81a9-11eb-8e88-aad7ddfa698b.png)

* sub3 gets displaced
* there shouldn't be an interruption in numbering form `1. hello 5` to `1. hello 6`

but this could potentially be addressed in future PR, wdyt ?

Reply:
Everything is working as expected here.

- sub3 is not displaced. It's just parsed as normal paragraph according to RST specification because there is zook5 without indent after it
- the same here: "1. hello 6" was parsed as a normal paragraph and zook6 became a block quote because it's indented by 1

And the warning is issued in both cases, so it looks ok to me.

rst2html.py does the same thing if you change `1`s to `#`s.

Also note that presence of auto-numerator `1` does not mean that it's a real "markdown mode".

Reply:
I think it's possible to implement also markdown-style enumeration list parsing because it's basically just relaxation of the RST kind. But it would be another task.